client/login/magic-login/index.jsx

```diff
@@ -4,130 +4,88 @@
-import EmailedLoginLinkSuccessfully from './emailed-login-link-successfully';
-import EmailedLoginLinkExpired from './emailed-login-link-expired';
-import HandleEmailedLinkForm from './handle-emailed-link-form';
 import RequestLoginEmailForm from './request-login-email-form';
-import PageViewTracker from 'lib/analytics/page-view-tracker';
 import GlobalNotices from 'components/global-notices';
+import Gridicon from 'gridicons';
 
 class MagicLogin extends React.Component {
 	static propTypes = {
+		// mapped to dispatch
 		hideMagicLoginRequestForm: PropTypes.func.isRequired,
-		magicLoginEmailAddress: PropTypes.string,
-		magicLoginEnabled: PropTypes.bool,
-		magicLoginView: PropTypes.string,
+		recordPageView: PropTypes.func.isRequired,
 		recordTracksEvent: PropTypes.func.isRequired,
-		showMagicLoginInterstitialPage: PropTypes.func.isRequired,
+
+		// mapped to state
+		showCheckYourEmail: PropTypes.bool.isRequired,
+
+		// From `localize`
 		translate: PropTypes.func.isRequired,
 	};
 
 	onClickEnterPasswordInstead = event => {
 		event.preventDefault();
-		this.props.recordTracksEvent( 'calypso_login_enter_password_instead_click' );
+		this.props.recordTracksEvent( 'calypso_login_email_link_page_click_back' );
 
 		page( login( { isNative: true } ) );
 	};
 
-	magicLoginMainContent() {
+	render() {
 		const {
-			magicLoginView,
-			magicLoginEmailAddress,
+			showCheckYourEmail,
+			translate,
 		} = this.props;
 
-		switch ( magicLoginView ) {
-			case LINK_EXPIRED_PAGE:
-				this.props.recordTracksEvent( 'calypso_login_magic_link_expired_link_view' );
-				return <EmailedLoginLinkExpired />;
-			case CHECK_YOUR_EMAIL_PAGE:
-				this.props.recordTracksEvent( 'calypso_login_magic_link_link_sent_view' );
-				return <EmailedLoginLinkSuccessfully emailAddress={ magicLoginEmailAddress } />;
-			case INTERSTITIAL_PAGE:
-				this.props.recordTracksEvent( 'calypso_login_magic_link_interstitial_view' );
-				return <HandleEmailedLinkForm />;
-		}
-	}
-
-	componentWillMount() {
-		const {
-			magicLoginEnabled,
-			queryArguments,
-		} = this.props;
+		this.props.recordPageView( '/log-in/link', 'Login > Link' );
 
-		if ( magicLoginEnabled && queryArguments && queryArguments.action === 'handleLoginEmail' ) {
-			this.props.showMagicLoginInterstitialPage();
-		}
-	}
+		const footer = ! showCheckYourEmail && (
+			<div className="magic-login__footer">
+				<a href="#" onClick={ this.onClickEnterPasswordInstead }>
+					<Gridicon icon="arrow-left" size={ 18 } /> { translate( 'Enter a password instead' ) }
+				</a>
+			</div>
+		);
 
-	render() {
-		const {
-			magicLoginView,
-			translate,
-		} = this.props;
 		return (
-			<Main className={ {
-				'magic-login': true,
-				'magic-login__request_link': ! magicLoginView || magicLoginView === REQUEST_FORM,
-			} }>
-				<PageViewTracker path="/log-in/link" title="Login > Link" />
-
+			<Main className={ classNames( 'magic-login', {
+				'magic-login__request_link': ! showCheckYourEmail,
+			} ) }>
```

Comment:
We could make this block a little bit easier to read by moving this `classNames()` call above in a variable.